float32 vx
float32 vy
float32 vyaw
float32 height